### DISCUSSION TITLE
Help me to understand nested generics please

### DISCUSSION THREAD

**Discussion by lua-rocks** [teal-language/tl#498] [Q&A] [answered]
This is example from tutorial, but I changed value **3** to **"wtf"** and I expected to get an error, because X in this case is a number and "wtf" is a string. But there are no errors. Why?

```lua
local record Tree<X>
  {Tree<X>}
  item: X
end

local t: Tree<number> = {
  item = 1,
  { item = 2 },
  { item = "wtf", { item = 4 } },
}
```

Also VSCode says that `t[2].item` is `number`  🙈

![screenshot](https://user-images.githubusercontent.com/82697845/147654510-fafa7526-95ad-4791-9135-f23df5915a9e.png)

**Comment by hishamhm** [comment, marked as answer]
This was a bug and it's now fixed in #499! Thanks!
